serde/src/ser/mod.rs

```diff
@@ -77,345 +77,357 @@ pub trait Serialize {
 /// state object, as it is expected that the user will not modify it. Due to the generic nature
 /// of the `Serialize` impls, modifying the object is impossible on stable Rust.
 pub trait Serializer {
-    /// The error type that can be returned if some error occurs during serialization.
+    /// Trickery to enforce correct use of the `Serialize` trait. Every
+    /// `Serializer` should set `Ok = ()`.
+    type Ok;
```

Comment:
Oh I should fix the comment. It actually turns out to be super useful. Check out https://github.com/serde-rs/json/commit/dc53c603b5fea6a49c68aca25183dd676a788e68.